specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5214 @@
+            },
+            "swaggerUri": {
+              "description": "The Uri for sending swagger requests.",
+              "type": "string"
+            },
+            "modelConfigMap": {
+              "description": "Details on the models and configurations.",
+              "type": "object",
+              "additionalProperties": {
+                "type": "object"
+              }
+            },
+            "models": {
+              "description": "The list of models.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Model"
+              }
+            },
+            "environmentImageRequest": {
+              "description": "The Environment, models and assets used for inferencing.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/EnvironmentImageRequest"
+                }
+              ]
+            },
+            "vnetConfiguration": {
+              "description": "The virtual network configuration.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/VnetConfiguration"
+                }
+              ]
+            },
+            "encryptionProperties": {
+              "description": "The encryption properties.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/EncryptionProperties"
+                }
+              ]
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "ACI"
+    },
+    "ContainerResourceRequirements": {
+      "description": "The resource requirements for the container (cpu and memory).",
+      "type": "object",
+      "properties": {
+        "cpu": {
+          "format": "double",
+          "description": "The number of CPU cores on the container.",
+          "type": "number",
+          "example": 4
+        },
+        "memoryInGB": {
+          "format": "double",
+          "description": "The amount of memory on the container in GB.",
+          "type": "number",
+          "example": 64
+        },
+        "gpu": {
+          "format": "int32",
+          "description": "The number of GPU cores in the container.",
+          "type": "integer"
+        },
+        "fpga": {
+          "format": "int32",
+          "description": "The number of FPGA PCIE devices exposed to the container. Must be multiple of 2.",
+          "type": "integer"
+        }
+      }
+    },
+    "ModelDataCollection": {
+      "description": "The Model data collection properties.",
+      "type": "object",
+      "properties": {
+        "eventHubEnabled": {
+          "description": "Option for enabling/disabling Event Hub.",
+          "type": "boolean"
+        },
+        "storageEnabled": {
+          "description": "Option for enabling/disabling storage.",
+          "type": "boolean"
+        }
+      }
+    },
+    "VnetConfiguration": {
+      "type": "object",
+      "properties": {
+        "vnetName": {
```

Comment:
you may know the destination RG but someone randomly looking at this resource has no way of knowing where this vnet is.  Please use the full resource ID in your ARM APIs

Reply:
all our customers using other clients(sdk, cli and portal) are used to populating vnet and subnet names instead of resourceIds. Supporting resourceIds instead of individual params requires backend changes followed by SDP, that can take few weeks, this will be addressed in the next iteration of the preview release.